includes/admin/upgrades/class-give-updates.php

```diff
@@ -316,38 +316,58 @@ public function __redirect_admin() {
 	 * @since  2.0.1
 	 * @access public
 	 *
+	 * @param bool $force
+	 *
 	 * @return bool
 	 */
-	public function __pause_db_update() {
+	public function __pause_db_update( $force = false ) {
 		// Bailout.
 		if (
-			wp_doing_ajax() ||
-			! isset( $_GET['page'] ) ||
-			'give-updates' !== $_GET['page'] ||
-			! isset( $_GET['give-pause-db-upgrades'] ) ||
-			self::$background_updater->is_paused_process()
+			! $force &&
+			(
+				wp_doing_ajax() ||
+				! isset( $_GET['page'] ) ||
+				'give-updates' !== $_GET['page'] ||
+				! isset( $_GET['give-pause-db-upgrades'] ) ||
+				self::$background_updater->is_paused_process()
+			)
+
 		) {
 			return false;
 		}
 
 		$batch = self::$background_updater->get_all_batch();
 
-		if ( ! empty( $batch ) ) {
-			update_option( 'give_paused_batches', $batch,  'no' );
-			delete_option( $batch->key );
-			delete_site_transient( self::$background_updater->get_identifier() . '_process_lock' );
-			wp_clear_scheduled_hook( self::$background_updater->get_cron_identifier() );
+		// Bailout: if batch is empty
+		if( empty( $batch ) ) {
+			return false;
+		}
 
-			Give()->logs->add( 'Update Pause', print_r( $batch, true ), 0, 'update' );
+		// Remove cache.
+		Give_Background_Updater::flush_cache();
 
-			/**
-			 * Fire action when pause db updates
-			 *
-			 * @since 2.0.1
-			 */
-			do_action( 'give_pause_db_upgrade', $this );
+		// Do not stop background process immediately if task running.
+		if( ! $force && self::$background_updater->is_process_running()  ) {
```

Comment:
@DevinWalker Previously we are pausing background updates immediately which can cause of stalled process due to the race condition.

Instead of stopping background update immediately, now it will stop after finishing current task.